misk-aws-dynamodb-testing/src/test/resources/moviestestmodule-common.yaml

```diff
@@ -0,0 +1,3 @@
+docker_dynamodb_config:
+  host: "localhost"
+  port: 8000
```

Comment:
I don't think this needs to be configurable. Rather, it should automatically discover an unused port and use it. E.g. something like this https://gist.github.com/vorburger/3429822

Reply:
the main downside to a random port is debugging using the aws cli. although if you don't use a random port, you could easily create a flaky tests if multiple CI branches are running at the same host.

I think we should have a convention that a random port is used in CI and a static known (although uncommon to avoid local conflicts) for local development.

Reply:
If we assign a random unused port to this DynamoDB local container, do we risk occupying a port required by another test, such as 27000?

Launching a container takes a few seconds. It'd be great if we can reuse the same DynamoDB local container for multiple tests.

Reply:
> If we assign a random unused port to this DynamoDB local container, do we risk occupying a port required by another test, such as 27000?

Possible, not likely. Feel free to choose a static port too, as long as it doesn't collide.

> Launching a container takes a few seconds. It'd be great if we can reuse the same DynamoDB local container for multiple tests.

Now that I look at this PR in detail, I think something got lost. `StartDatabaseService` is unnecessary. The Hibernate StartDatabaseService is a pattern I'd like to get rid of but just haven't gotten around to. The right way to instantiate a fake DynamoDB is to add this to your test. This already works

```
@MiskExternalDependency private val dockerDynamoDb = DockerDynamoDb
```

But this still has the issue that you can no longer share the same container across test runs because the port name is not hard-coded. There are ways to work around this but I think it's a bit complicated and we are better off just using static ports for now.